src/libs/sfdk/virtualmachine.cpp

```diff
@@ -196,6 +203,7 @@ void VirtualMachine::setHeadless(bool headless)
     Q_D(VirtualMachine);
     if (d->headless == headless)
         return;
+    QTC_ASSERT(d->features & OptionalHeadless, return);
```

Comment:
No it shouldn't. If it goes higher, we get always the assert failures because BuildEnginePrivate::fromMap() always calls setHeadless(). And there's nothing wrong with that, as it never changes the setting.

Reply:
hmm. I suppose I should just change the fromMap() to check the feature flags...